src/devices/machine/tc009xlvc.cpp

```diff
@@ -14,186 +14,133 @@
 #include "emu.h"
 #include "machine/tc009xlvc.h"
 
+#include "emupal.h"
 #include "screen.h"
 
 
 DEFINE_DEVICE_TYPE(TC0091LVC, tc0091lvc_device, "tc009xlvc", "Taito TC0091LVC")
 
-
-READ8_MEMBER(tc0091lvc_device::tc0091lvc_paletteram_r)
+u8 tc0091lvc_device::rom_r(offs_t offset)
 {
-	return m_palette_ram[offset & 0x1ff];
+	if ((offset & 0x6000) == 0x6000) // bankswitched area
+		offset = (*m_rom_bank << 13) | (offset & 0x1fff);
+
+	return m_rom[offset & m_rom.mask()];
 }
 
-WRITE8_MEMBER(tc0091lvc_device::tc0091lvc_paletteram_w)
+void tc0091lvc_device::vram_w(offs_t offset, u8 data)
 {
-	m_palette_ram[offset & 0x1ff] = data;
+	// note, the way tiles are addressed suggests that 0x0000-0x3fff of this might be usable,
+	//       but we don't map it anywhere, so the first tiles are always blank at the moment.
+	if (offset < 0x4000) // <0x4000 area is not mapped?
```

Comment:
Accessing like this means that the VRAM internal space needs to be routed in a different address_space, namely for a pratical reason, every time you're accessing like this you're making a new check, giving a performance penalty.